docs/src/solvers/nonlinear_system_solvers.md

```diff
@@ -0,0 +1,163 @@
+  - [`SimpleTrustRegion()`](@ref): A dogleg trust-region Newton method. Improved globalizing
+    stability for more robust fitting over basic Newton methods, though potentially with a
+    cost.
+  - [`SimpleDFSane()`](@ref): A low-overhead implementation of the df-sane method for
+    solving large-scale nonlinear systems of equations.
+  - [`SimpleHalley()`](@ref): A low-overhead implementation of the Halley method. This is a
+    higher order method and thus can converge faster to low tolerances than a Newton method.
+    Requires higher order derivatives, so best used when automatic differentiation is
+    available.
+
+!!! note
+    
+    When used with certain types for the states `u` such as a `Number` or `StaticArray`,
+    these solvers are very efficient and non-allocating. These implementations are thus
+    well-suited for small systems of equations.
+
+### SteadyStateDiffEq.jl
+
+SteadyStateDiffEq.jl uses ODE solvers to iteratively approach the steady state. It is a
+very stable method for solving nonlinear systems with stable equilibrium points, though
+often more computationally expensive than direct methods.
+
+  - [`DynamicSS()`](@ref): Uses an ODE solver to find the steady state. Automatically
+    terminates when close to the steady state.
+  - [`SSRootfind()`](@ref): Uses a NonlinearSolve compatible solver to find the steady
+    state.
+
+### NLsolve.jl
+
+This is a wrapper package for importing solvers from NLsolve.jl into the SciML interface.
+
+  - [`NLsolveJL()`](@ref): A wrapper for
+    [NLsolve.jl](https://github.com/JuliaNLSolvers/NLsolve.jl)
+
+Submethod choices for this algorithm include:
+
+  - `:anderson`: Anderson-accelerated fixed-point iteration
+  - `:newton`: Classical Newton method with an optional line search
+  - `:trust_region`: Trust region Newton method (the default choice)
+
+### MINPACK.jl
+
+MINPACK.jl methods are good for medium-sized nonlinear solves. It does not scale due to
+the lack of sparse Jacobian support, though the methods are very robust and stable.
+
+  - [`CMINPACK()`](@ref): A wrapper for using the classic MINPACK method through
+    [MINPACK.jl](https://github.com/sglyon/MINPACK.jl)
+
+Submethod choices for this algorithm include:
+
+  - `:hybr`: Modified version of Powell's algorithm.
+  - `:lm`: Levenberg-Marquardt.
+  - `:lmdif`: Advanced Levenberg-Marquardt
+  - `:hybrd`: Advanced modified version of Powell's algorithm
+
+### Sundials.jl
+
+Sundials.jl are a classic set of C/Fortran methods which are known for good scaling of the
+Newton-Krylov form. However, KINSOL is known to be less stable than some other
+implementations.
+
+  - [`KINSOL()`](@ref): The KINSOL method of the SUNDIALS C library
+
+### SIAMFANLEquations.jl
+
+SIAMFANLEquations.jl is a wrapper for the methods in the SIAMFANLEquations.jl library.
+
+  - [`SIAMFANLEquationsJL()`](@ref): A wrapper for using the methods in
+    [SIAMFANLEquations.jl](https://github.com/ctkelley/SIAMFANLEquations.jl)
```

Comment:
Some of the newer wrappers are missing from this list. SpeedMapping.jl, LeastSquaresOptim.jl, FastLevenburgMarquedt, etc.

Reply:
Linked them from here to avoid repeating the text.